### ISSUE TITLE
typecheck task fails with typescript variant of create-mc-app

### ISSUE THREAD

**Issue by lwilkins** [commercetools/merchant-center-application-kit#3018]
**Describe the issue**

With an otherwise un modified version of the typescript starter from create-mc-app, the built in typecheck task fails:
```
node_modules/@formatjs/ecma402-abstract/types/date-time.d.ts:113:18 - error TS2430: Interface 'DateTimeFormat' incorrectly extends interface 'Omit<DateTimeFormat, "resolvedOptions">'.
  The types returned by 'formatRangeToParts(...)' are incompatible between these types.
    Type 'IntlDateTimeFormatPart[]' is not assignable to type 'DateTimeRangeFormatPart[]'.
      Type 'IntlDateTimeFormatPart' is not assignable to type 'DateTimeRangeFormatPart'.
        Types of property 'source' are incompatible.
          Type 'RangePatternType | undefined' is not assignable to type '"shared" | "startRange" | "endRange"'.
            Type 'undefined' is not assignable to type '"shared" | "startRange" | "endRange"'.

113 export interface DateTimeFormat extends Omit<Intl.DateTimeFormat, 'resolvedOptions'> {
                     ~~~~~~~~~~~~~~

node_modules/@formatjs/intl/src/types.d.ts:19:26 - error TS2339: Property 'ids' does not exist on type 'Message'.

19 } ? FormatjsIntl.Message['ids'] : string;
                            ~~~~~

node_modules/@formatjs/intl/src/types.d.ts:22:29 - error TS2339: Property 'locale' does not exist on type 'IntlConfig'.

22 } ? FormatjsIntl.IntlConfig['locale'] : string;
                               ~~~~~~~~
```

**To Reproduce**

1. Run `npx @commercetools-frontend/create-mc-app@latest test-app-name --template=starter-typescript` accepting default options.
2. Run `yarn typecheck`

**Expected behavior**

To pass with no errors.

**Additional context**

I've tried to fix myself and downgrading TypeScript to 4.8.4 fixes two of the errors. However the other I'm not sure. Reading around, I believe this is routed in upgrading `react-intl` and fixable by https://github.com/commercetools/merchant-center-application-kit/pull/2760. However, that pulls in other complexities.

It's hard to run TypeScript in my CI environment without this being fixed as Typescript always returns a non zero exit code.

**Comment by Rhotimee** [comment]
Hi @lwilkins, thank you for submitting this issue. 
We will look into it.

**Comment by emmenko** [comment]
Hey @lwilkins 

Sorry for the inconvenience. Until we upgrade `react-intl` and other dependencies we can't do much about it.

In our repository here we had the same issue and were able to solve it by [patching the types](https://github.com/commercetools/merchant-center-application-kit/blob/main/.yarn/patches/%40formatjs-intl-npm-2.2.1-1251194b73.patch) of `react-intl`.

Furthermore, you might also need to set a resolution on `@formatjs/ecma402-abstract` package, as we are doing here:
https://github.com/commercetools/merchant-center-application-kit/blob/697eeaabb3abe8de012ae22b70f0e60cd7e66466/package.json#L160

Please let us know if this helps fixing the type errors.

**Comment by lwilkins** [comment]
Hi both.

Thanks, I copied your local patch to `@formatjs/ecma402-abstract` and I had to create a similar one for `@formatjs/intl`. This fixed the TypeScript errors for me, but I would expect this sort of thing to be solved out of the box however that comes. Some note in the documentation would have been good if not otherwise fixable in code automatically.

I'll close the issue as I'm otherwise good now.
